### DISCUSSION TITLE
Update with old values

### DISCUSSION THREAD

**Discussion by sualko** [mikro-orm/mikro-orm#5862] [Q&A]
TL;DR; mikro-orm overwrites some fields with outdated values in a long running task. Why?

We have a long running task (several hours) where we get a list of items at the start. After a few hours mikro-orm writes back some fields to the database with the values it read before. It doesn't write back all fields and not for all items. We can just remove the items from the identity manager, but we would like to understand why this happens. Does someone has an idea.

To visualize it a bit better I will try to create a timeline below.

```
Task 1: const items = await em.find(Item, {}); // [A, B, C]    and e.g. A.a = 1
Task 1: doing some stuff
Task 2: starts and modifies some values (different entity manager, maybe different machine) e.g. A.a = 2
Task 1: is still doing stuff
Task 2: writes A.a = 1, before a transaction starts with a native update
```

Those values which will be updated are not touched by task 1, but all of them are either of type date or json.

```
@Property({
    columnType: 'timestamp',
    length: 6,
    nullable: true,
})
a: Date | null = null;

@Property({ type: 'json', defaultRaw: `('[]')` })
b: Scope[] = [];
```

I was not able to create a reproducible test case. My best guess was that something with the comparison of date and json is not working, but in the code it looks like `valueOf` is used which should be fine for dates. Does someone has an idea? I can fix it probably by disabling the identity map for those calls, but I would like to know what is going on.
